### PR TITLE
New report player button in player menu.

### PR DESCRIPTION
**Pull request by ConorGriffin37**
Add a new report button in the player menu to allow users to easily report someone.

### CODE REVIEW

File: CoreScriptsRoot/Modules/Settings/Pages/Players.lua [hunk 1]
```diff
@@ -73,14 +72,114 @@ local function Initialize()
...
+
+		if friendLabelParent then
 			-- remove any previous friend status labels
-			for _, item in pairs(playerLabel:GetChildren()) do
-				if item and item.Name == 'FriendStatus' then
+			for _, item in pairs(friendLabelParent:GetChildren()) do
+				if item and item.Name == "FriendStatus" then
 					if GuiService.SelectedCoreObject == item then
-						selectionFound = nil
+						friendSelectionFound = nil
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Settings/Pages/Players.lua]: would this cause issues with navigation if I only have a gamepad?

File: CoreScriptsRoot/Modules/Settings/Pages/Players.lua [hunk 2]
```diff
@@ -93,67 +192,56 @@ local function Initialize()
...
-				local addFriendFunc = function()
-					if friendLabel and friendLabelText and friendLabelText.Text ~= '' then
-						friendLabel.ImageTransparency = 1
-						friendLabelText.Text = ''
-						if localPlayer and player then
-							localPlayer:RequestFriendship(player)
-						end
+			if enableReportPlayer then
+				local friendLabel = nil
+				local wasIsPortrait = nil
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Settings/Pages/Players.lua]: this never seems to be set

File: CoreScriptsRoot/Modules/Settings/SettingsHub.lua [hunk 3]
```diff
@@ -457,9 +470,9 @@ local function CreateSettingsHub()
 					inviteToGameFunc, {Enum.KeyCode.ButtonX}
 				)
 				if RunService:IsStudio() then
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Settings/SettingsHub.lua]: how is this code even run? is xbox and studio?

File: CoreScriptsRoot/Modules/Settings/SettingsHub.lua [hunk 4]
```diff
@@ -945,16 +958,26 @@ local function CreateSettingsHub()
 
 		-- make sure page is visible
 		this.Pages.CurrentPage = pageToSwitchTo
-		this.Pages.CurrentPage:Display(this.PageView, skipAnimation)
+		this.Pages.CurrentPage:Display(this.PageViewInnerFrame, skipAnimation)
 		this.Pages.CurrentPage.Active = true
 
 		local pageSize = this.Pages.CurrentPage:GetSize()
 		this.PageView.CanvasSize = UDim2.new(0,pageSize.X,0,pageSize.Y)
+		if this.PageView.CanvasSize.Y.Offset > this.PageView.AbsoluteSize.Y then
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Settings/SettingsHub.lua]: does this need to be flagged?

File: CoreScriptsRoot/Modules/Settings/SettingsHub.lua (hunk 3 above)
Comment (ConorGriffin37) [inline CoreScriptsRoot/Modules/Settings/SettingsHub.lua]: It is for the xbox emulation in studio.

**Pull request merged**